src/Raven.Server/Documents/Handlers/BatchHandler.cs

```diff
@@ -248,20 +254,17 @@ private async Task HandleClusterTransaction(DocumentsOperationContext context, M
                 };
             }
 
-            // wait for the command to be applied on this node
-            await ServerStore.WaitForCommitIndexChange(RachisConsensus.CommitIndexModification.GreaterOrEqual, result.Index);
-
             var array = new DynamicJsonArray();
+            
+            // wait for the command to be applied on this node,
+            // canceling will be done through the cts and not with a timeout
+            await ServerStore.WaitForCommitIndexChange(RachisConsensus.CommitIndexModification.GreaterOrEqual, result.Index, Timeout.InfiniteTimeSpan, cts.Token);
```

Comment:
The default HTTP timeout is 12 hours, no?

Reply:
same reason as above